### ISSUE TITLE
Reexport `createBullBoard`, `BullAdapter` and `BullMQAdapter` in server adapters

### ISSUE THREAD

**Issue by mmachatschek** [felixmosh/bull-board#500]
It would be nice if `@bull-board/express` and the other server adapters would export `createBullBoard`, `BullAdapter` and `BullMQAdapter` from the `@bull-board/api` package.

This would come with the benefit to manage a little less dependencies.

The hello world example then could look like this:

```diff
const express = require('express');
const Queue = require('bull');
const QueueMQ = require('bullmq');
- const { createBullBoard } = require('@bull-board/api');
- const { BullAdapter } = require('@bull-board/api/bullAdapter');
- const { BullMQAdapter } = require('@bull-board/api/bullMQAdapter');
- const { ExpressAdapter } = require('@bull-board/express');
+ const { ExpressAdapter, createBullBoard, BullAdapter, BullMQAdapter } = require('@bull-board/express');

const someQueue = new Queue('someQueueName', {
  redis: { port: 6379, host: '127.0.0.1', password: 'foobared' },
}); // if you have a special connection to redis.
const someOtherQueue = new Queue('someOtherQueueName');
const queueMQ = new QueueMQ('queueMQName');

const serverAdapter = new ExpressAdapter();
serverAdapter.setBasePath('/admin/queues');

const { addQueue, removeQueue, setQueues, replaceQueues } = createBullBoard({
  queues: [new BullAdapter(someQueue), new BullAdapter(someOtherQueue), new BullMQAdapter(queueMQ)],
  serverAdapter: serverAdapter,
});

const app = express();

app.use('/admin/queues', serverAdapter.getRouter());

// other configurations of your server
  
app.listen(3000, () => {
  console.log('Running on 3000...');
  console.log('For the UI, open http://localhost:3000/admin/queues');
  console.log('Make sure Redis is running on port 6379 by default');
});
```

**Comment by felixmosh** [comment]
Hi @mmachatschek I've thought about this once, but decided that it is better to be explicit, serverAdapter comes from the server adapter nothing more.

In addition, the hello world example is not a real world example, since you probably won't use 2 type of queues (`bull` & `bullMQ`).

So the code is minimal as,
```diff
const express = require('express');
const Queue = require('bull');
const QueueMQ = require('bullmq');
const { createBullBoard } = require('@bull-board/api');
- const { BullAdapter } = require('@bull-board/api/bullAdapter');
const { BullMQAdapter } = require('@bull-board/api/bullMQAdapter');
const { ExpressAdapter } = require('@bull-board/express');

- const someQueue = new Queue('someQueueName', {
-   redis: { port: 6379, host: '127.0.0.1', password: 'foobared' },
- }); // if you have a special connection to redis.
- const someOtherQueue = new Queue('someOtherQueueName');
const queueMQ = new QueueMQ('queueMQName');

const serverAdapter = new ExpressAdapter();
serverAdapter.setBasePath('/admin/queues');

const { addQueue, removeQueue, setQueues, replaceQueues } = createBullBoard({
  queues: [
-    new BullAdapter(someQueue), 
-    new BullAdapter(someOtherQueue), 
     new BullMQAdapter(queueMQ)],
  serverAdapter: serverAdapter,
});

const app = express();

app.use('/admin/queues', serverAdapter.getRouter());

// other configurations of your server
  
app.listen(3000, () => {
  console.log('Running on 3000...');
  console.log('For the UI, open http://localhost:3000/admin/queues');
  console.log('Make sure Redis is running on port 6379 by default');
});
```

**Comment by mmachatschek** [comment]
> The problem with this is that every change in API should be done in all of the adapters... this makes the code highly coupled.

Isn't it already coupled? The adapters already depend on a specific version of the api package, exporting them via the server adapter would make sense to ensure the correct version of the underlying functions is used on the client code. Current implementations would not be affected by the change.

> What is the issue of installing both of the packages?

There is no immediate issue here, the suggested solution makes it more maintainable and ensures that the server adapter gehts the correct data from the api package.
Future me wants to avoid a mistake of upgrading one of the bullboard depencencies without the other one and raise issue that possible don't exist if the correct version would be used.

**Comment by felixmosh** [comment]
Thanks for the PR.
It released in `v4.8.0`

**Comment by mmachatschek** [comment]
Thanks for the super fast response time 🚀
